src/main/java/core/basesyntax/strategy/DataConverterImpl.java

```diff
@@ -0,0 +1,21 @@
+package core.basesyntax.strategy;
+
+import core.basesyntax.model.FruitTransaction;
+import core.basesyntax.model.FruitTransaction.Action;
+import java.util.ArrayList;
+import java.util.List;
+
+public class DataConverterImpl implements DataConverter {
+    public List<FruitTransaction> convertToTransaction(List<String> dateList) {
+        List<FruitTransaction> result = new ArrayList<>();
+        for (String string : dateList) {
+            String[] stringArray = string.split(",");
+            Action action = Action.fromCode(stringArray[0]);
+            String fruit = stringArray[1];
+            int quantity = Integer.parseInt(stringArray[2]);
```

Comment:
It's a good practice to check the length of the `stringArray` after splitting to make sure it contains the expected number of elements before accessing them. This will prevent `ArrayIndexOutOfBoundsException` if the input data is malformed.